base/loading.jl

```diff
@@ -3333,6 +3341,65 @@ function resolve_depot(inc::AbstractString)
     return :no_depot_found
 end
 
+"""
+    Base.RelocPath(path::AbstractString)
+
+A type to represent a relocatable path.
+
+Requires `path` to be located within one of `DEPOT_PATH` upon construction.
+
+An error is thrown if relocation fails.
+
+# Example
+```jldoctest
+julia> path1 = joinpath(mktempdir(), "foo"); touch(path1); # set up a file called foo
+
+julia> pushfirst!(DEPOT_PATH, dirname(path1));
+
+julia> relocpath = Base.RelocPath(path1);
+
+julia> String(relocpath) == path1
+true
+
+julia> path2 = joinpath(mktempdir(), "foo"); touch(path2); # set up another foo
+
+julia> pushfirst!(DEPOT_PATH, dirname(path2));
+
+julia> String(relocpath) == path2
+true
+
+julia> path1 != path2
+true
+```
+"""
+struct RelocPath
+    subpath::String
+    function RelocPath(path::AbstractString)
+        depot, _ = replace_depot_path_impl(path)
+        if isnothing(depot)
+            error("Failed to locate $(path) in any of DEPOT_PATH.")
+        end
+        subpath = replace(path, depot => ""; count=1)
+        return new(subpath)
+    end
+end
+
+function String(r::RelocPath)
+    for d in DEPOT_PATH
+        if isdirpath(d)
+            d = dirname(d)
+        end
+        path = string(d, r.subpath)
+        if ispath(path)
+            return path
+        end
+    end
+    error("Failed to relocate @depot$(r.subpath) in any of DEPOT_PATH.")
+end
+
+function show(io::IO, r::RelocPath)
+    print(io, string("RelocPath(\"@depot", r.subpath, "\")"))
+end
```

Comment:
This is pretty inaccurate, since it doesn't support `@depot`
```suggestion
```

Reply:
How about we allow RelocPath to accept paths that are literally `@depot/foo/bar`?

Reply:
Also, as far as the original `show` method goes, I think that the meaning of `@depot/` is reasonably easy to understand here.

Reply:
> This is pretty inaccurate, since it doesn't support @depot

But it does behave like a `@depot`. (it behaves more like `include_dependency` than `include` to be precise)

What is inaccurate is that `show` assumes that nobody tinkered with `DEPOT_PATH` since construction.